cimgui/build.gradle.kts

```diff
@@ -1,341 +1,335 @@
+	if (!useSingleTarget || HostManager.hostIsMac) macosX64()
+
+	jvm {
+		withJava()
+		compilations {
+			"main" {
+				defaultSourceSet {
+					kotlin.srcDir("src/jvmMain/java")
+					kotlin.srcDir("src/jvmMain/kotlin")
+				}
+				dependencies {
+					implementation(kotlin("stdlib-jdk8"))
+				}
+			}
+			"test" {
+				dependencies {
+					implementation(kotlin("test"))
+					implementation(kotlin("test-junit"))
+				}
+			}
+		}
+	}
+
+	sourceSets {
+		commonMain {
+			dependencies {
+				implementation(kotlin("stdlib-common"))
+			}
+		}
+		commonTest {
+			dependencies {
+				implementation(kotlin("test-common"))
+				implementation(kotlin("test-annotations-common"))
+			}
+		}
+	}
+
+	targets.withType<KotlinNativeTarget> {
+		val libDir = buildDir.resolve("lib").resolve(targetName)
+
+		val objDir = libDir.resolve("obj")
+		val staticLibFile = libDir.resolve("libimgui.a")
+
+		val objFiles = objFileNames.map { objDir.resolve(it) }
+		val compileImGui = tasks.register<Exec>("compileImGuiFor$targetName") {
+			dependsOn(cloneCimgui)
+			onlyIf { HostManager().isEnabled(konanTarget) }
+			doFirst { mkdir(objDir) }
+
+			inputs.files(sourceFiles)
+			outputs.files(objFiles)
+
+			environment("PATH", "$llvmBinFolder;${System.getenv("PATH")}")
+			if (konanTarget == KonanTarget.MACOS_X64) {
+				environment(
+					"CPATH",
+					"/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include"
+				)
+			}
+
+			executable = llvmBinFolder.resolve("clang++").absolutePath
+
+			val targetInfo = targetInfoMap.getValue(konanTarget)
+			if (HostManager.host != konanTarget) args("-target", targetInfo.targetName)
+			args("--sysroot=${targetInfo.sysRoot}")
+			args(targetInfo.clangArgs)
+			args(
+				"-c", "-w",
+				"-I${imguiDir}", "-I${cimguiOutput}", "-I${cimguiDir}",
+				"-working-directory", objDir.absolutePath
+			)
+			args(sourceFiles.map { it.absolutePath })
+		}
+		val archiveImGui = tasks.register<Exec>("archiveImGuiFor$targetName") {
+			dependsOn(compileImGui)
+			onlyIf { HostManager().isEnabled(konanTarget) }
+
+			inputs.files(objFiles)
+			outputs.file(staticLibFile)
+
+			environment("PATH", "$llvmBinFolder;${System.getenv("PATH")}")
+			executable = llvmBinFolder.resolve("llvm-ar").absolutePath
+			args("rc", staticLibFile.absolutePath)
+			args(objFiles.map { it.absolutePath })
+		}
+
+		compilations {
+			"main" {
+				cinterops {
+					create("cimgui") {
+						includeDirs(cimguiOutput, imguiDir)
+						tasks.named(interopProcessingTaskName) {
+							dependsOn(cloneCimgui)
+						}
+						compileImGui {
+							// This is to ensure native toolchain is downloaded before compiling with it.
+							dependsOn(tasks.named(interopProcessingTaskName))
+						}
+					}
+				}
+
+				compileKotlinTask.dependsOn(archiveImGui)
+				kotlinOptions {
+					freeCompilerArgs = listOf("-include-binary", staticLibFile.absolutePath)
+				}
+			}
+		}
+	}
+
+	val jvmTargets = listOfNotNull(
```

Comment:
This can be `listOf(HostManager.host)` for the `useSingleTarget` case.

Reply:
implemented